dss/api/search.py

```diff
@@ -70,21 +70,15 @@ def post(json_request_body: dict, replica: str, per_page: int, _scroll_id: typin
         } for hit in page['hits']['hits']]
 
         # TODO: (tsmith12) if page returns 0 hits, then all results have been found. delete search id
-        # TODO: (tsmith12) if all results found return request.code.ok.
         if len(result_list) < per_page:
```

Comment:
I would consider doing this:

```
if len(result_list) < per_page:
    result_code = requests.codes.ok
else:
    result_code = requests.codes.partial
response = make_response(jsonify({'es_query': es_query, 'results': result_list}), result_code)
```

Reply:
The problem is I only want to add the `Link` in the header when a next url is returned. I intentionally isolated `response.headers['Link'] = build_link_header({next_url: {"rel": "next"}})` into the `else` to prevent `Link` from appearing in the header.